config/v1/types_infrastructure.go

```diff
@@ -144,6 +148,18 @@ type GCPPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// BareMetalPlatformStatus holds the current status of the BareMetal infrastructure provider.
+type BareMetalPlatformStatus struct {
+	// api_vip holds the Virtual IP (VIP) that can be used to contact the API
+	APIVIP string `json:"api_vip"`
+
+	// ingress_vip holds the VIP for Ingress
+	IngressVIP string `json:"ingress_vip"`
+
+	// dns_vip holds the VIP for DNS
```

Comment:
Indicate what names we expect this DNS resolve.  If we try to control entries in here, indicate that tool.